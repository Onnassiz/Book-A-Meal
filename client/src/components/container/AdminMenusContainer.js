import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getProfile } from '../../reducer/profile/actionsDispatcher';
import { getMeals } from '../../reducer/meals/actionsDispatcher';
import { postMenu, getUserMenus, deleteMenuById, updateMenu, getMealsInMenu } from '../../reducer/menus/actionsDispatcher';
import AdminMenus from '../presentation/AdminMenus';

export const mapStateToProps = state => ({
  menus: state.menus,
  meals: state.meals,
  user: state.user,
  profile: state.profile,
  formState: state.formState,
});

export function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    postMenu,
    deleteMenuById,
    getUserMenus,
    getMealsInMenu,
    getMeals,
    getProfile,
    updateMenu,
  }, dispatch);
}

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(AdminMenus));
