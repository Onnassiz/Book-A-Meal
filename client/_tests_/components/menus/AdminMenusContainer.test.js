import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import AdminMenusContainer, { mapStateToProps, mapDispatchToProps } from '../../../src/components/container/AdminMenusContainer';

const mockStore = configureStore();

const store = mockStore({
  profile: {},
  user: {},
  formState: { isLoading: false },
  menus: {},
  meals: {},
});

const initialState = {
  profile: {},
  user: {},
  formState: { isLoading: false },
};

const wrapper = shallow(<AdminMenusContainer store={store} />);

describe('<AdminMenusContainer />', () => {
  it('Should render the AdminMenus container successfully', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getProfile();
    expect(mapStateToProps(initialState).profile).toEqual({});
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
