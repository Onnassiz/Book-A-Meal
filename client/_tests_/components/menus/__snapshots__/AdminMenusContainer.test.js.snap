// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminMenusContainer /> Should render the AdminMenus container successfully 1`] = `
<Route
  render={[Function]}
/>
`;
